### ISSUE TITLE
Unit test spec file missing mock @Inject MsalGuardConfiguration part

### ISSUE THREAD

**Issue by zhshqzyc** [AzureAD/microsoft-authentication-library-for-js#4934]
### Core Library

MSAL.js v2 (@azure/msal-browser)

### Core Library Version

2.14

### Wrapper Library

MSAL Angular (@azure/msal-angular)

### Wrapper Library Version

2

### Description

I want to do unit test for app.component. In the constructor there are many injection  but your spec file doesn't include all. So it is hard to follow them. I know most of them are similar, but for the @InjectToken part, I hope you can add the mock code.

`constructor(
    @Inject(MSAL_GUARD_CONFIG) private msalGuardConfig: MsalGuardConfiguration,
    private authService: MsalService,
    private msalBroadcastService: MsalBroadcastService
  ) `

### Relevant Code Snippets

```javascript
How to mock the @Inject part, can you add it in your spec file so we can follow it in our unit test? 
There is an example https://stackoverflow.com/questions/56250733/how-to-test-component-with-a-inject-text-in-the-constructor-in-angular. Which is string but here is the MsalGuardConfiguration type, can we mock it?

@Inject(MSAL_GUARD_CONFIG) private msalGuardConfig: MsalGuardConfiguration,
```

### Source

External (Customer)

**Comment by bmahall** [comment]
@zhshqzyc I am currently looking into this  and will get back as soon as I have more information .

**Comment by bmahall** [comment]
@zhshqzyc Can you try the following code to mock @Inject MsalGuardConfiguration in the concerned unit test case.
```

import { Injectable } from '@angular/core';


function MSALInstanceFactory(): IPublicClientApplication {
...
}

describe('AppComponent', () => {
  let MsalGuardConfigMock = {  
//provide the desired mock configuration here
//adding one for example usage
    interactionType: InteractionType.Popup ,
    authRequest: Request,
    loginFailedRoute: "/"
  };

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        RouterTestingModule
      ],
      declarations: [
        AppComponent
      ],
      providers: [
        MsalService,
        {
          provide: MSAL_INSTANCE,
          useFactory: MSALInstanceFactory
        },
        { //this will replace MsalGuardConfiguration with MsalGuardConfigMock
          provide: MSAL_GUARD_CONFIG,
          useValue: MsalGuardConfigMock
        }
      ]
    }).compileComponents();
  }));

...

//Example test case reference
  it('should use the msalguardconfig mock',()=> {
     expect(MsalGuardConfigMock.authRequest).toBeDefined();
     ...
  });



```
Please let me know in case of any further questions. Thanks

**Comment by zhshqzyc** [comment]
In the app component constructor, we still have `private msalBroadcastService: MsalBroadcastService,`

Can you also mock it? Because there is the error
`NullInjectorError No provider for MsalBroadcastService`

**Comment by ghost** [comment]
This issue requires attention from the MSAL.js team and has not seen activity in **5 days**. @bmahall please follow up.

**Comment by ghost** [comment]
This issue requires attention from the MSAL.js team and has not seen activity in **5 days**. @bmahall please follow up.

**Comment by peterzenz** [comment]
We will track this item internally for a future sprint.  Thanks.
